### ISSUE TITLE
Robovac 30C having decoding issues

### ISSUE THREAD

**Issue by vmartinv** [CodeFoodPixels/robovac#59]
After upgrading from 1.2.0 to 1.2.2, the vacuum became unavailable.

These are my logs:
```
2024-02-28 00:34:36.338 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Processing queue. Current length: 683
2024-02-28 00:34:36.339 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Sending to 55103800e098069fb103 (192.168.0.32:6668): Message(0x9, b'', 0, <Device 55103800e098069fb103 (192.168.0.32:6668)>)
2024-02-28 00:34:36.345 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Received message from 55103800e098069fb103 (192.168.0.32:6668): Message(0x9, b'', 0, None)
2024-02-28 00:34:39.517 ERROR (MainThread) [custom_components.robovac.tuyalocalapi] 'utf-8' codec can't decode byte 0xd5 in position 17: invalid continuation byte
2024-02-28 00:34:47.676 ERROR (MainThread) [custom_components.robovac.tuyalocalapi] 'utf-8' codec can't decode byte 0xdf in position 5: invalid continuation byte
```

Now after some time is showing:
```
2024-02-28 11:16:33.887 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Processing queue. Current length: 1
2024-02-28 11:16:33.888 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Sending to 55103800e098069fb103 (:6668): Message(0x7, {'devId': '55103800e098069fb103', 'uid': '', 't': 1709118993, 'dps': {'5': 'auto'}}, 53838084, <Device 55103800e098069fb103 (:6668)>)
2024-02-28 11:16:33.889 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Connecting to 55103800e098069fb103 (:6668)
2024-02-28 11:16:33.890 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Retrying send due to error. Connection to 55103800e098069fb103 (:6668) failed: [Errno 111] Connection refused
2024-02-28 11:16:34.141 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Sending to 55103800e098069fb103 (:6668): Message(0x7, {'devId': '55103800e098069fb103', 'uid': '', 't': 1709118993, 'dps': {'5': 'auto'}}, 53838084, <Device 55103800e098069fb103 (:6668)>)
2024-02-28 11:16:34.142 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Connecting to 55103800e098069fb103 (:6668)
2024-02-28 11:16:34.143 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Retrying send due to error. Connection to 55103800e098069fb103 (:6668) failed: [Errno 111] Connection refused
2024-02-28 11:16:34.394 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Sending to 55103800e098069fb103 (:6668): Message(0x7, {'devId': '55103800e098069fb103', 'uid': '', 't': 1709118993, 'dps': {'5': 'auto'}}, 53838084, <Device 55103800e098069fb103 (:6668)>)
2024-02-28 11:16:34.395 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Connecting to 55103800e098069fb103 (:6668)
2024-02-28 11:16:34.396 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] 1 failures. Most recent: Connection to 55103800e098069fb103 (:6668) failed: [Errno 111] Connection refused
2024-02-28 11:16:34.706 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Processing queue. Current length: 1
2024-02-28 11:16:34.706 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Sending to 55103800e098069fb103 (:6668): Message(0x7, {'devId': '55103800e098069fb103', 'uid': '', 't': 1709118994, 'dps': {'5': 'auto'}}, 53838939, <Device 55103800e098069fb103 (:6668)>)
2024-02-28 11:16:34.707 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Connecting to 55103800e098069fb103 (:6668)
2024-02-28 11:16:34.708 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Retrying send due to error. Connection to 55103800e098069fb103 (:6668) failed: [Errno 111] Connection refused
2024-02-28 11:16:34.959 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Sending to 55103800e098069fb103 (:6668): Message(0x7, {'devId': '55103800e098069fb103', 'uid': '', 't': 1709118994, 'dps': {'5': 'auto'}}, 53838939, <Device 55103800e098069fb103 (:6668)>)
2024-02-28 11:16:34.960 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Connecting to 55103800e098069fb103 (:6668)
2024-02-28 11:16:34.961 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Retrying send due to error. Connection to 55103800e098069fb103 (:6668) failed: [Errno 111] Connection refused
2024-02-28 11:16:35.212 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Sending to 55103800e098069fb103 (:6668): Message(0x7, {'devId': '55103800e098069fb103', 'uid': '', 't': 1709118994, 'dps': {'5': 'auto'}}, 53838939, <Device 55103800e098069fb103 (:6668)>)
2024-02-28 11:16:35.212 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Connecting to 55103800e098069fb103 (:6668)
2024-02-28 11:16:35.213 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] 2 failures. Most recent: Connection to 55103800e098069fb103 (:6668) failed: [Errno 111] Connection refused
```

**Comment by CodeFoodPixels** [comment]
This looks like it's struggling to connect to the vacuum, could you try removing the vacuum and reconfiguring please?

**Comment by vmartinv** [comment]
> This looks like it's struggling to connect to the vacuum, could you try removing the vacuum and reconfiguring please?

Hum I did, it got fixed. But then it went into error again. Restarting solved the issue. I'll see if it keeps happening.

**Comment by lukemorrigan-pp** [comment]
The `utf-8` thing seems to happen sometimes because of a change in the local key or something like that. I've never encountered it myself but I've read about it on the localtuya issues

**Comment by CodeFoodPixels** [comment]
The `utf-8` thing seems to happen sometimes because of a change in the local key or something like that. I've never encountered it myself but I've read about it on the localtuya issues

**Comment by FuturamaPhill** [comment]
I've got a similar issue with a brand new Eufy 15C MAX

`2024-02-28 19:08:44.403 WARNING (MainThread) [custom_components.robovac.vacuum] Update errored. Current update failure count: 1. Reason: Timed out waiting for response to sequence number 1536048335
2024-02-28 19:09:43.954 WARNING (MainThread) [custom_components.robovac.vacuum] Update errored. Current update failure count: 2. Reason: Timed out waiting for response to sequence number 1536108335

2024-02-28 19:16:43.435 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Processing queue. Current length: 25483
2024-02-28 19:16:43.435 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Sending to bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668): Message(0x9, b'', 0, <Device bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668)>)
2024-02-28 19:16:43.435 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Retrying send due to error. Connection to bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668) failed: [Errno 104] Connection reset by peer

2024-02-28 19:16:43.544 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Disconnected from bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668)
2024-02-28 19:16:43.545 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Connection reset
2024-02-28 19:16:43.686 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Sending to bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668): Message(0x9, b'', 0, <Device bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668)>)
2024-02-28 19:16:43.686 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Connecting to bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668)
2024-02-28 19:16:43.693 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Disconnected from bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668)
2024-02-28 19:16:43.870 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Processing queue. Current length: 38079
2024-02-28 19:16:43.870 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Sending to bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668): Message(0x9, b'', 0, <Device bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668)>)
2024-02-28 19:16:43.870 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Connecting to bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668)
2024-02-28 19:16:43.874 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Disconnected from bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668)
2024-02-28 19:16:44.042 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Processing queue. Current length: 41287
2024-02-28 19:16:44.042 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Sending to bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668): Message(0x9, b'', 0, <Device bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668)>)
2024-02-28 19:16:44.042 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Connecting to bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668)
2024-02-28 19:16:44.093 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Disconnected from bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668)
2024-02-28 19:16:45.109 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Processing queue. Current length: 52832
2024-02-28 19:16:45.109 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Sending to bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668): Message(0x9, b'', 0, <Device bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668)>)
2024-02-28 19:16:45.109 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Connecting to bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668)
2024-02-28 19:16:45.288 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Disconnected from bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668)
2024-02-28 19:16:45.565 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Processing queue. Current length: 79398
2024-02-28 19:16:45.566 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Sending to bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668): Message(0x9, b'', 0, <Device bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668)>)
2024-02-28 19:16:45.566 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Connecting to bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668)
2024-02-28 19:16:45.735 DEBUG (MainThread) [custom_components.robovac.tuyalocalapi] Disconnected from bf4b0ace39a5f9f09ajma7 (192.168.0.144:6668)
`
![image](https://github.com/CodeFoodPixels/robovac/assets/39866073/6e34a743-4cd4-4771-91ad-bb253b6a60a3)

Tried reinstalling integration and bouncing HA but still seems to struggle to connect, able to control the vac fine through the app. Any thoughts on what else to try?
